File: week6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

def main ():

    #get text from user
     text = get_string("Text: ")

     no_of_letters = count_letters(text)
     no_of_words = count_words(text)
     no_of_scentences = count_scentences(text)


     l = no_of_letters* (100/ no_of_words)

     s = no_of_scentences *(100/ no_of_words)

     index =round((0.0588 * l) - (0.296 * s) - 15.8)

     if index > 16:

        print("Grade 16+")

     elif index < 1:
        print("Before Grade 1")
     else:
         print(f"Grade {index}")



#function to count no of letters
def count_letters(text):
  counter=0
  for c in text :
     # condition to count alpha only
     if c != 32 and  (c >= "a" and c <= 'z') or (c >= "A" and c <= "Z"):
       counter +=1


  return counter

#function to count no of words
def count_words( text):


     counter=1
     for c in text :
        #condition to check spaces
        if c == " ":
            counter += 1
     return counter

# function to count the no of scentences
def count_scentences (text ):
    counter=0
    for c in text :
        #condition to check . ! ?
        if c == "." or c == "?" or c == "!" :
           counter +=1
    return counter

if __name__ == "__main__":
    main()

File: week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>
int count_letters(string text);
int count_words(string text) ;
int count_scentences(string text);

int main(void)
{
    //get text from user
    string text = get_string("Text: ");

    int no_of_letters = count_letters(text) ;
    int no_of_words = count_words(text);
    int no_of_scentences = count_scentences(text);


    float l = (((float)no_of_letters / no_of_words) * 100);

    float s = (((float)no_of_scentences / no_of_words) * 100) ;

    float index = (0.0588 * l) - (0.296 * s) - 15.8 ;

    if ((int) round(index) > 15)
    {
         printf("Grade 16+\n");
    }
    else if ((int) round(index) < 2)
    {
         printf ("Before Grade 1\n");
    }
    else
    {
         printf ("Grade %i\n", (int) round(index));
    }


}



//function to count no of letters
int count_letters(string text)
{
    int i = 0 ;
    int counter = 0;
    while (text[i] != '\0')
    {
        // condition to count alpha only 
        if (text[i] != 32 && isalpha(text[i]) != 0)
        {
            counter ++;
        }
        i ++;
    }
    return counter ;
}
//function to count no of words
int count_words(string text)
{
    int i = 0 ;
    int counter = 0;
    while (text[i] != '\0')
    {
        //condition to check spaces
        if (text[i] == 32)
        {
           counter ++;
        }
        i ++;
    }
    counter ++ ;
    return counter ;
}
// function to count the no of scentences
int count_scentences (string text )
{
    int i = 0 ;
    int counter = 0;
    while (text[i] != '\0')
    {
        //condition to check . ! ?
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            counter ++ ;
        }
        i ++ ;
    }
    return counter ;
}



